### PR TITLE
feat!: add computed_members, project, condition

### PR DESCRIPTION
**Pull request by sfh-darknet**

### CODE REVIEW

File: versions.tf [hunk 1]
```diff
@@ -1,10 +1,10 @@
 terraform {
-  required_version = ">= 0.14, < 2.0"
+  required_version = "~> 1.0"
```
Comment (waxb) [inline versions.tf]: these changes should go into the module update commit

File: variables.tf [hunk 2]
```diff
@@ -29,6 +29,17 @@ variable "members" {
   }
 }
 
+variable "computed_members_map" {
+  type        = map(string)
+  description = "(Optional) A map of members to replace in 'var.members' to handle terraform computed values. Will be ignored when policy bindings are used."
+  default     = {}
+
+  validation {
+    condition     = alltrue([for k, v in var.computed_members_map : can(regex("^(allUsers|allAuthenticatedUsers|(user|serviceAccount|group|domain):)", v))])
```
Comment (waxb) [inline variables.tf:38]: please update the condition according to the [resource's documentation](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/artifact_registry_repository_iam#member/members). 
aka. copy paste as i did before, what is wrong that i know now thanks to @mariux :)
is this a minus beer?

**Review by waxb — CHANGES_REQUESTED** [review]
- please move [module config section](https://github.com/mineiros-io/terraform-google-artifact-registry-repository-iam#module-configuration) under [main config](https://github.com/mineiros-io/terraform-google-artifact-registry-repository-iam#main-resource-configuration) in the readme
- please have the breaking change (dropping support for provider version) into changelog
- please clean up commits to: `chore: changelog, feat: features, chore!: update from module template`

File: main.tf [hunk 3]
```diff
@@ -5,7 +5,7 @@ resource "google_artifact_registry_repository_iam_binding" "binding" {
   repository = var.repository
   location   = var.location
   role       = var.role
-  members    = var.members
```
Comment (waxb) [inline main.tf:8]: please add the `dynamic "condition" {}` block

File: main.tf [hunk 4]
```diff
@@ -18,7 +18,7 @@ resource "google_artifact_registry_repository_iam_member" "member" {
   repository = var.repository
   location   = var.location
   role       = var.role
-  member     = each.value
```
Comment (waxb) [inline main.tf:21]: please add the `dynamic "condition" {}` block

File: variables.tf (hunk 2 above)
Comment (waxb) [reply variables.tf:38]: no worries, actually there is a beer counter from @mariux for every time i made this mistake

Comment (sfh-darknet) [reply variables.tf]: hahahaha add me up for 1 beer then

**Review by mariux — APPROVED** [review]
lgtm

File: CHANGELOG.md [hunk 5]
```diff
@@ -7,6 +7,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 ## [Unreleased]
 
+### Added
+
+- Add variable `condition`
+- Add support for `computed_members`
+
+### Removed
+
+- BREAKING CHANGE: dropping support for `provider version`
```
Comment (mariux) [inline CHANGELOG.md:17]:
```suggestion
- BREAKING CHANGE: dropping support for Terraform Google Provider before `4.0`.
```

**Review by mariux — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 5 above)
Comment (sfh-darknet) [reply CHANGELOG.md]: Fixed and added new lines where missing in my editor ;)

File: test/unit-minimal/main.tf [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  source = "../.."
+
+  # add all required arguments
+
+  repository = "unit-minimal-${local.random_suffix}"
+  location   = "us-central1"
+
+  # add all optional arguments that create additional/extended resources
+
+  role = "roles/artifactregistry.reader"
```
Comment (waxb) [inline test/unit-minimal/main.tf:11]: please only have the required arguments at unit-minimal

File: test/unit-minimal/main.tf [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
```
Comment (waxb) [inline test/unit-minimal/main.tf:65]: please only have the required arguments at unit-minimal

**Review by waxb — CHANGES_REQUESTED** [review]

File: README.tfdoc.hcl [hunk 8]
```diff
@@ -226,42 +251,69 @@ section {
...
+        All attributes of the created 'iam_binding' or 'iam_member' or 'iam_policy' resource according to the mode."
       END
-    }
-  }
+        }
+      }
 
-  section {
-    title = "External Documentation"
+      section {
```
Comment (mariux) [inline README.tfdoc.hcl:292]: indentation looks wrong now?

**Review by mariux — CHANGES_REQUESTED** [review]

File: README.tfdoc.hcl (hunk 8 above)
Comment (sfh-darknet) [reply README.tfdoc.hcl:292]: terradoc would cry if it was? it was crying before and fixed it

**Review by soerenmartius — APPROVED** [review]
lgtm! good job @sfh-darknet

**Review by mariux — APPROVED** [review]
lgtm

**Pull request merged**